src/Tickets/Commerce/Cart.php

```diff
@@ -659,5 +669,29 @@ public function process( array $data = [] ) {
 		// Before we start we clear the existing cart.
 		return $this->get_repository()->process( $data );
 	}
-
+	
+	/**
+	 * Get the total of the cart.
+	 *
+	 * @since TBD
+	 *
+	 * @return null|float
+	 */
+	public function get_cart_total() {
+		if ( null !== $this->cart_total ) {
+			return $this->cart_total;
+		}
+		
+		$items = $this->get_items_in_cart( true );
+		
+		if ( empty( $items ) ) {
+			return null;
+		}
+		
+		foreach ( $items as $item ) {
+			$this->cart_total += $item['sub_total']->get_string();
+		}
+		
+		return $this->cart_total;
+	}
```

Comment:
This method shouldn't live in the cart controller, it should live in the `Cart/Unmanaged_Cart.php` the controller should use the `$this->get_repository()` to get the method from the correct cart.

Also add the method to the interface.

Reply:
The Cart total should be calculated the same way irrespective of Cart Type.

Just like we have https://github.com/the-events-calendar/event-tickets/blob/0ab55ce3636af51e9ec04a00502ee0a8fb2e71c7/src/Tickets/Commerce/Cart.php#L353

That's why it is placed within the cart controller. I have now moved both them into an abstract to keep common methods in a place.